### ISSUE TITLE
Hardcoded length restrictions for topo file paths

### ISSUE THREAD

**Issue by tovogt** [clawpack/geoclaw#479]
We have a setup where paths to our geoclaw run directories can become really long. In GeoClaw, there is a hardcoded restriction to 150 characters for topo file paths in https://github.com/clawpack/geoclaw/blob/dbcf70fbd9bc7dd604349d643d1f7c38575779a7/src/2d/shallow/topo_module.f90#L16
For our purposes, something like 200 or 250 characters would be convenient. Is there anything that prevents you from increasing that number? I'm not so familiar with Fortran: Is there an efficient way to dynamically allocate the string to avoid that problem altogether?

**Comment by mandli** [comment]
Nope, in fact you will probably find that there are a few places where it's longer and shorter.  In this case, since this is a type definition we just need to know how long the string is and we can go from there so we do not even need to specify a length given that we will have just read it in.

**Comment by tovogt** [comment]
Just for completeness, to raise this restriction, we have to increase the numbers in the following three lines:
https://github.com/clawpack/geoclaw/blob/dbcf70fbd9bc7dd604349d643d1f7c38575779a7/src/2d/shallow/topo_module.f90#L16
https://github.com/clawpack/geoclaw/blob/dbcf70fbd9bc7dd604349d643d1f7c38575779a7/src/2d/shallow/topo_module.f90#L431
https://github.com/clawpack/geoclaw/blob/dbcf70fbd9bc7dd604349d643d1f7c38575779a7/src/2d/shallow/topo_module.f90#L723
(And probably the analogous lines for `dtopo`.)

So, would you agree to increase these numbers to a higher value? Or would you rather reimplement everything with dynamic string lengths as obtained from the read operations?

**Comment by mandli** [comment]
I would have no objection.  Might as well raise them to something nice and `2^n` like `1024`.

**Comment by mandli** [comment]
I should also say though there's a number of places where paths are input, might be nice to define a parameter that just dimensions all of these somewhere centralized.  @mjberger is there a common way to do this for path strings in Fortran that you are aware of?
